resource/abstractions/auditing_rules.md

```diff
@@ -0,0 +1,478 @@
+```
+
+## `ResourceAdmin.clear_all_audit_rules()`
+
+```python
+def clear_all_audit_rules(self, resource: str, class_name: str) -> Union[dict, bytes]:
+```
+
+#### 📄 Description
+
+Clear any rules defined within the **Audit Rules** for a general resource profile.
+
+#### 📥 Parameters
+* `resource`<br>
+  The **general resource profile** for which RACF should change the audit rules.
+
+* `class_name`<br>
+  The name of the **class** the general resource profile being changed belongs to.
+
+#### 📤 Returns
+* `Union[dict, bytes]`<br>
+  Returns a **Security Result Steps dictionary** or **Concatenated Security Request XML bytes** if the `ResourceAdmin.generate_requests_only` class attribute is set to `True`.
+
+#### ❌ Raises
+* `SecurityRequestError`<br>
+  Raises `SecurityRequestError` when the **SAF Return Code** of a **Security Result** is equal to `4`.
+* `DownstreamFatalError`<br>
+  Raises `DownstreamFatalError` when the **SAF Return Code** of a **Security Result** is greater than `4`.
+* `AlterOperationError`<br>
+  Raises `AlterOperationError` when the **general resource profile** cannot be altered because it does **NOT** exist.
+
+#### 💻 Example
+
+###### Python REPL
+```python
+>>> from pyracf import ResourceAdmin
+>>> resource_admin = ResourceAdmin()
+>>> resource_admin.clear_all_audit_rules("TESTING","ELIJTEST")
+{"step1":{"securityResult":{"resource":{"name":"TESTING","class":"ELIJTEST","operation":"set","requestId":"ResourceRequest","info":["Definition exists. Add command skipped due  to precheck option"],"commands":[{"safReturnCode":0,"returnCode":0,"reasonCode":0,"image":"RALTER  ELIJTEST             (TESTING)  AUDIT( NONE    )","messages":["ICH11009I RACLISTED PROFILES FOR ELIJTEST WILL NOT REFLECT THE UPDATE(S) UNTIL A SETROPTS REFRESH IS ISSUED."]}]},"returnCode":0,"reasonCode":0}}}
+```
+
+###### Security Result Steps Dictionary as JSON
+```json
+{
+  "step1": {
+    "securityResult": {
+      "resource": {
+        "name": "TESTING",
+        "class": "ELIJTEST",
+        "operation": "set",
+        "requestId": "ResourceRequest",
+        "info": [
+          "Definition exists. Add command skipped due  to precheck option"
+        ],
+        "commands": [
+          {
+            "safReturnCode": 0,
+            "returnCode": 0,
+            "reasonCode": 0,
+            "image": "RALTER  ELIJTEST             (TESTING)  AUDIT( NONE    )",
+            "messages": [
+              "ICH11009I RACLISTED PROFILES FOR ELIJTEST WILL NOT REFLECT THE UPDATE(S) UNTIL A SETROPTS REFRESH IS ISSUED."
+            ]
+          }
+        ]
+      },
+      "returnCode": 0,
+      "reasonCode": 0
+    }
+  }
+}
+```
+
+## `ResourceAdmin.overwrite_audit_by_attempt()`
+
+```python
+def overwrite_audit_by_attempt(
+    self,
+    resource: str,
+    class_name: str,
+    successes: Union[str, None] = None,
+    failure: Union[str, None] = None,
+    all: Union[str, None] = None,
+) -> Union[dict, bytes]:
+```
+
+#### 📄 Description
```

Comment:
Add a notes annotation to each alter/overwrite function describing what valid **Access Level** and **Access Attempts** are for each function?

Reply:
Added the following notes:

```
{: .note }
> _Valid values for **Access Attempts** are limited to the **Access Levels** `alter`, `control`, `read`, and `update`._
> _You may not specify the same **Access Level** for 2+ **Access Attempts** on a single call of `overwrite_audit_by_attempt`._

{: .note }
> _Valid values for **Access Levels** are limited to the **Access Attempts** `success`, `failure`, and `all`._
```

Reply:
Move these section to directly below the `#### 📄 Description` heading. Also pay close attention to line breaks.

Reply:
```markdown
{: .note }
> _Valid values for **Access Attempts** are limited to the **Access Level** values of `alter`, `control`, `read`, and `update`._

&nbsp;

{:.warning}
> _Using an **Access Level** value more than once is **NOT** allowed._
```

```markdown
{: .note }
> _Valid values for **Access Levels** are limited to the **Access Attempt** values of `success`, `failure`, and `all`._
```

Reply:
changed.

Reply:
Surround annotations with a line break:

```markdown
&nbsp;

{: .note }
> _Valid values for **Access Attempts** are limited to the **Access Level** values of `alter`, `control`, `read`, and `update`._

&nbsp;

{:.warning}
> _Using an **Access Level** value more than once is **NOT** allowed._

&nbsp;
```

```markdown
&nbsp;

{: .note }
> _Valid values for **Access Levels** are limited to the **Access Attempt** values of `success`, `failure`, and `all`._

&nbsp;
```

Reply:
ah, yes, that makes sense. fixed.